src/test/java/de/rub/nds/modifiablevariable/integer/ModifiableIntegerTest.java

```diff
@@ -9,84 +9,247 @@
+        byte[] expected = new byte[] {0x01, 0x02}; // Big-endian, 2 bytes
+        assertArrayEquals(expected, integer1.getByteArray(2));
+
+        // Test with different sizes
+        assertArrayEquals(new byte[] {0x00, 0x00, 0x01, 0x02}, integer1.getByteArray(4));
+        assertArrayEquals(new byte[] {0x02}, integer1.getByteArray(1));
     }
 
     /** Test of validateAssertions method, of class ModifiableInteger. */
     @Test
     public void testValidateAssertions() {
+        // No assertions set - should be valid
         assertTrue(integer1.validateAssertions());
+        assertTrue(nullInteger.validateAssertions());
+
+        // Set matching assertion
+        integer1.setAssertEquals(2);
+        assertTrue(integer1.validateAssertions());
+
+        // Set non-matching assertion
         integer1.setAssertEquals(3);
         assertFalse(integer1.validateAssertions());
+
+        // Test with modification
+        integer1 = new ModifiableInteger(10);
+        integer1.setAssertEquals(11);
+        assertFalse(integer1.validateAssertions());
+
+        VariableModification<Integer> modifier = IntegerModificationFactory.add(1);
+        integer1.setModifications(modifier);
+        assertTrue(integer1.validateAssertions());
+    }
+
+    /** Test of isOriginalValueModified method, of class ModifiableInteger. */
+    @Test
+    public void testIsOriginalValueModified() {
+        // No modification
+        assertFalse(integer1.isOriginalValueModified());
+
+        // With modification that changes value
+        VariableModification<Integer> modifier = IntegerModificationFactory.add(1);
+        integer1.setModifications(modifier);
+        assertTrue(integer1.isOriginalValueModified());
+
+        // With modification that doesn't change value
+        modifier = IntegerModificationFactory.add(0);
+        integer2.setModifications(modifier);
+        assertFalse(integer2.isOriginalValueModified());
+
+        // Null original value - if null, then getValue() is also null, so not modified
+        assertFalse(nullInteger.isOriginalValueModified());
+
+        // Following the implementation in ModifiableInteger.isOriginalValueModified(),
+        // a null originalValue will always return false for isOriginalValueModified()
+        // even when the getValue() returns a non-null value
+        ModifiableInteger nullIntWithExplicit = new ModifiableInteger((Integer) null);
+        nullIntWithExplicit.setModifications(IntegerModificationFactory.explicitValue(5));
+
+        // Verify getValue() is 5
+        assertEquals(Integer.valueOf(5), nullIntWithExplicit.getValue());
+
+        // The implementation returns false when originalValue is null, even if getValue() != null
+        // This is because the check in isOriginalValueModified() is:
+        //   getOriginalValue() != null && getOriginalValue().compareTo(getValue()) != 0;
+        assertFalse(nullIntWithExplicit.isOriginalValueModified());
```

Comment:
this is odd - maybe we should change the behavior when the original value is null